dlt-database/entity/0001-init_db/ConfirmedTransaction.ts

```diff
@@ -0,0 +1,57 @@
+import {
+  Entity,
+  PrimaryGeneratedColumn,
+  Column,
+  CreateDateColumn,
+  ManyToOne,
+  JoinColumn,
+  OneToOne,
+} from 'typeorm'
+import { Decimal } from 'decimal.js-light'
+
+import { DecimalTransformer } from '../../src/typeorm/DecimalTransformer'
+import { Account } from './Account'
+import { TransactionDraft } from './TransactionDraft'
+
+@Entity('confirmed_transactions')
+export class ConfirmedTransaction {
+  @PrimaryGeneratedColumn('increment', { unsigned: true, type: 'bigint' })
+  id: number
+
+  @OneToOne(() => TransactionDraft, (draft) => draft.confirmedTransaction)
+  @JoinColumn({ name: 'transaction_draft_id' })
+  transactionDraft: TransactionDraft
+
+  @Column({ name: 'transaction_draft_id', type: 'int', unsigned: true })
+  transactionDraftId: number
+
+  @Column({ type: 'bigint' })
+  nr: number
+
+  @Column({ type: 'binary', length: 48 })
+  runningHash: Buffer
+
+  @ManyToOne(() => Account, (account) => account.confirmedTransactions)
+  @JoinColumn({ name: 'account_id' })
+  account: Account
+
+  @Column({ name: 'account_id', type: 'int', unsigned: true })
+  accountId: number
+
+  @Column({
+    name: 'account_balance',
+    type: 'decimal',
+    precision: 40,
+    scale: 20,
+    nullable: false,
+    default: 0,
+    transformer: DecimalTransformer,
+  })
+  accountBalance: Decimal
```

Comment:
The balance will be calculated for the date the transaction was confirmed so confirmed_at = balance_date

Reply:
i think that will lead to a timeshifting and in consequence to different balance and decay values.
On my opinion the balance_date will be fixed with the timestamp of the transaction itself and all other calculations and verifications should use this prefixed date. In consequence it doesn't matter when the verification take place it will calculate with the fixed balance_date and will result always in the same values of decay and balance.

Reply:
It is to prevent timeshifting. The confirmed_at is the iota milestone timestamp. This ensures that all transactions in the network work with a single time reference. It is to prevent this:
![Byzantine Fault Tolerance](https://hedera.com/learning/hedera-hashgraph/what-is-asynchronous-byzantine-fault-tolerance-abft)
So my prefixed date is the timestamp of the iota milestone in which the transaction was ordered from iota. So every listen node server get's exact the same data. Even when the backend server clocks are not aligned or been tampered with.
Show me that you solution will get the same ordering of transactions and have the same balance for the same date.
You know the decay function is a function. It describe the decay value on a timeline. I can calculate the decay value for every given time. That's the reason why I usually don't store the decay value for a point in time. I can easly calculate it with a subtraction between two transactions or with the formula for every other point in time between them.